### PR TITLE
Gci 1451 create chprd records

### PR DESCRIPTION
**Pull request by rjadhav-ch**
Implements saving of MID order details to CHPRD `ORDERHEADER` and `ORDERDETAIL` tables.

### CODE REVIEW

File: src/main/java/uk/gov/companieshouse/chd/order/api/controller/MissingImageDeliveriesController.java [hunk 1]
```diff
@@ -48,7 +56,10 @@ public MissingImageDeliveriesController(final CreateItemRequestValidator createM
             return ResponseEntity.status(BAD_REQUEST).body(new ApiError(BAD_REQUEST, errors));
         }
 
+        MissingImageDeliveriesRequest midRequest = mapper.mapMissingImageDeliveriesRequest(midDTO);
```
Comment (VaughanJackson123) [inline src/main/java/uk/gov/companieshouse/chd/order/api/controller/MissingImageDeliveriesController.java:59]: Not sure about the name `MissingImageDeliveriesRequest` here. Typically wouldn't we use the term `request` to refer to what is passed in as a controller method parameter?

File: src/main/resources/application.properties [hunk 2]
```diff
@@ -1,2 +1,23 @@
...
+spring.datasource.url: ${CHPRD_DATASOURCE_URL}
+spring.datasource.username: ${CHPRD_SCHEMA_NAME}
+spring.datasource.password: ${CHPRD_PASSWORD}
+spring.datasource.driver.class: oracle.jdbc.driver.OracleDriver
+
+spring.jpa.hibernate.ddl-auto: none
+spring.jpa.show-sql: true
+spring.jpa.properties.hibernate.format_sql: true
+
+chprd.customer-id = CHPRD_CUSTOMER_ID
```
Comment (VaughanJackson123) [inline src/main/resources/application.properties:13]: Is this totally hardcoded then?

File: src/main/java/uk/gov/companieshouse/chd/order/api/controller/MissingImageDeliveriesController.java (hunk 1 above)
Comment (rjadhav-ch) [reply src/main/java/uk/gov/companieshouse/chd/order/api/controller/MissingImageDeliveriesController.java:59]: The DTO that this maps to is actually a request object. Without the `Request` prefix the POJO appears ambiguous.

Comment (VaughanJackson123) [reply src/main/java/uk/gov/companieshouse/chd/order/api/controller/MissingImageDeliveriesController.java:59]: As I say, it's more about code reader expectations when they read `...Request`. Not that important.

**Comment by VaughanJackson123** [comment]
Would it be possible to extend something like `MissingImageDeliveriesControllerIntegrationTest` to verify the database contents that result from handling the request?

**Review by nabeelamirch — APPROVED** [review]

File: src/main/resources/application.properties (hunk 2 above)
Comment (rjadhav-ch) [reply src/main/resources/application.properties]: Nice catch. This is incorrect. I have updated it now. 👍

File: pom.xml [hunk 3]
```diff
@@ -107,6 +122,15 @@
...
             <version>${environment-reader-library.version}</version>
         </dependency>
+
+        <dependency>
+            <groupId>com.h2database</groupId>
+            <artifactId>h2</artifactId>
+        </dependency>
+        <dependency>
+            <groupId>org.springframework.boot</groupId>
+            <artifactId>spring-boot-starter-test</artifactId>
```
Comment (nabeelamirch) [inline pom.xml:132]: Spring-boot-starter-test is defined twice.

**Review by VaughanJackson123 — DISMISSED** [review]

File: pom.xml (hunk 3 above)
Comment (rjadhav-ch) [reply pom.xml:132]: Nice catch. I'll fix this. Thanks 👍

**Review by VaughanJackson123 — APPROVED** [review]

**Review by nabeelamirch — APPROVED** [review]

**Pull request merged**